craterpy/tests/test_classes.py

```diff
@@ -97,3 +143,717 @@ def test_antimeridian_splitting(self):
+            )
+            np.testing.assert_array_almost_equal(
+                imported_cdb.rad.values, original_cdb.rad.values
+            )
+        finally:
+            # Clean up
+            if os.path.exists(tmp_path):
+                os.remove(tmp_path)
+
+    def test_to_geojson_with_custom_columns(self):
+        """Test reading a shapefile with non-standard column names."""
+        # Create a GeoDataFrame with non-standard column names
+        import geopandas as gpd
+
+        gdf = gpd.GeoDataFrame(
+            {
+                "latitude_deg": [0.0, 10.0],
+                "longitude_deg": [0.0, 20.0],
+                "diameter": [2.0, 4.0],  # Diameters in km
+                "crater_name": ["Crater A", "Crater B"],
+                "geometry": [Point(0, 0), Point(20, 10)],
+            },
+            crs="EPSG:4326",
+        )
+
+        import tempfile
+        import os
+
+        with tempfile.NamedTemporaryFile(
+            suffix=".geojson", delete=False
+        ) as tmp:
+            tmp_path = tmp.name
+
+        try:
+            # Save to file with non-standard column names
+            gdf.to_file(tmp_path, driver="GeoJSON")
+
+            # Read the file with non-standard column names
+            imported_cdb = CraterDatabase.read_shapefile(tmp_path, units="km")
+
+            # Verify data was preserved and correctly interpreted
+            self.assertEqual(len(imported_cdb.data), 2)
+
+            # Check that our read_shapefile method correctly identified the columns
+            np.testing.assert_array_almost_equal(
+                imported_cdb.lat.values, [0.0, 10.0]
+            )
+            np.testing.assert_array_almost_equal(
+                imported_cdb.lon.values, [0.0, 20.0]
+            )
+
+            # Since diameter was converted to radius in km and then to meters,
+            # we expect: diameter / 2 * 1000 = [1000.0, 2000.0]
+            np.testing.assert_array_almost_equal(
+                imported_cdb.rad.values, [1000.0, 2000.0]
+            )
+        finally:
+            # Clean up
+            if os.path.exists(tmp_path):
+                os.remove(tmp_path)
+
+    def test_read_shapefile_with_body_and_units(self):
+        """Test reading a shapefile with body and units information in the file."""
+        import geopandas as gpd
+        import tempfile
+        import os
+
+        # Create a GeoDataFrame with body and units info
+        gdf = gpd.GeoDataFrame(
+            {
+                "lat": [0.0, 10.0],
+                "lon": [0.0, 20.0],
+                "radius": [1.0, 2.0],
+                "name": ["Crater A", "Crater B"],
+                "body": [
+                    "Mars",
+                    "Mars",
+                ],  # Will NOT override user-provided body
+                "units": ["km", "km"],  # Will NOT override user-provided units
+                "geometry": [Point(0, 0), Point(20, 10)],
+            },
+            crs="EPSG:4326",
```

Comment:
I'd recommend not using epsg:4326 which is an earth projection here and elsewhere in the tests. since this is planetary focused and a lot of python geospatial packages work only for earth, it'll be better in general to test any planetary projection than an earth one for our purposes, e.g. `"IAU_2015:30100"`